contributing.md

```diff
@@ -31,6 +31,7 @@ You can also add new topics or update existing content:
 
 ## Guidelines for Writing Content
 
+* **Do include a table of contents.** A table of contents tells the reader what to expect, such as how long the article is or what content is covered. It breaks up writing into managable, easily digestable chunks and serves as a reference that makes it easier to jump around from section to section. In our case, we only show the table of contents for mobile users and for printing as Markbind already provides a navigation panel on the right side of every article. If you have programming experience, feel free to refer to the markdown file for [_Introduction to Go_](https://raw.githubusercontent.com/se-edu/learningresources/master/contents/go/Go.md) on how to implement this behavior. Otherwise, do include a table of contents in the document you send us.
```

Comment:
> It breaks up writing into managable, easily digestable chunks ...

This is not really a benefit of the ToC?

Reply:
> If you have programming experience, feel free to refer to the markdown file for [_Introduction to Go_](https://raw.githubusercontent.com/se-edu/learningresources/master/contents/go/Go.md) on how to implement this behavior. Otherwise, do include a table of contents in the document you send us.

Not sure what the expectation here. Why should they look at that file?

Reply:
> 
> This is not really a benefit of the ToC?

Hmm... the idea I was going for was that it gives a structure to the content that is presented, but I suppose this is a benefit of partitioning the content into sections rather than including a toc.

How about I change it to: "It also ~breaks up writing into managable, easily digestable chunks and~ serves as a reference that makes it easier to jump around from section to section."

>
> Not sure what the expectation here. Why should they look at that file?

I was thinking that everyone may have their own ideas of what a ToC would look like and that some contributors may not know much CSS. As such, they could just refer to that file for an idea of how to include their own ToC.

Reply:
> How about I change it to: "It also ~breaks up writing into managable, easily digestable chunks and~ serves as a reference that makes it easier to jump around from section to section."

Sure.

> I was thinking that everyone may have their own ideas of what a ToC would look like and that some contributors may not know much CSS. As such, they could just refer to that file for an idea of how to include their own ToC.

Let's give a clear instruction on how to include a ToC (can point to an example too), and yes, authors should not be required to know CSS.

Reply:
@damithc I've updated the guideline for table of contents. As an example, I included a link to a PR that adds a ToC to the Regex article.

I also added a guideline on using title case for headings so @Tejas2805 's work won't go to waste.

Reply:
Ah, I think I see why this instruction seems odd to me. It seems to assume that regular people who don't know programming will send articles to us and then one of us will help add the toc. That is not the case. As this is an SE resource, any author should be able to add the toc in the given format. There is no choice. So the instruction should be 'add a toc in this format.' period.

Reply:
Hmm.. that makes a lot of sense, but it seems to contradict what we have said in the earlier part of the document. The following seems to imply that people with little to no programming knowledge are welcome to send us their articles:

-----
## How to Contribute

You can report errors, send suggestions, ask questions. <trigger trigger="click" for="modal:contributing-contactInfo">Here</trigger> is how you can contact us.

You can also add new topics or update existing content:
* If you don't have programming experience, <trigger trigger="click" for="modal:contributing-contactInfo">send us</trigger> your document.
* If you have programming experience,
  * [Fork this repo](https://github.com/se-edu/learningresources/fork) and clone it to your Computer.
  * Install [MarkBind](https://markbind.github.io) in your Computer.
  * Post in [our issue tracker](https://github.com/nus-oss/learningresources/issues) an outline of your proposed contribution. Wait for our response.
  * After receiving the go-ahead from us, send your proposed contribution as a PR.
-----

Reply:
I went to the blame view to find who added that line. It was me! Don't know what I was thinking and have no memory of it either. Anyway, we can remove that too.